imports/ui/pages/professor/questions_library.jsx

```diff
@@ -36,21 +22,130 @@ class _QuestionsLibrary extends Component {
     this.state = {
       edits: {},
       selected: null,
-      questions: props.library,
-      limit: 11,
+      questions: props.questions,
       query: props.query,
-      questionMap: _(props.library).indexBy('_id'),
+      questionMap: _(props.questions).indexBy('_id'),
       resetSidebar: false // only to trigger prop update of side bar when creating new question and thus clear the filter (used as toggle)
     }
-
     if (this.props.selected) {
       if (this.props.selected in this.state.questionMap) this.state.selected = this.props.selected
     }
-
+  
+    this.convertField = this.convertField.bind(this)
+    this.exportQuestions = this.exportQuestions.bind(this)
+    this.importQuestions = this.importQuestions.bind(this)
     this.editQuestion = this.editQuestion.bind(this)
     this.questionDeleted = this.questionDeleted.bind(this)
+    this.deleteAllQuestions = this.deleteAllQuestions.bind(this)
     this.updateQuery = this.updateQuery.bind(this)
-    this.limitAndUpdate = _.throttle(this.limitAndUpdate.bind(this), 800)
+    this.approveQuestion = this.approveQuestion.bind(this)
+    this.deleteQuestion = this.deleteQuestion.bind(this)
+    this.makeQuestionPublic = this.makeQuestionPublic.bind(this)
+
+    Meteor.call('courses.getCourseCode', this.props.courseId, (e, c) => {
+      if (e) alertify.error('Cannot get course code')
+      else this.state.courseCode = c
+    })
+
+    Meteor.call('courses.hasAllowedStudentQuestions', this.props.courseId, (e, allowed) => {
+      if (e) alertify.error('Cannot get course permissions')
+      else this.state.allowedStudentQuestions = allowed
+    })
+  }
+
+  convertImageToBase64 (url, count, callback) {
+    let xhttp = new XMLHttpRequest()
+    xhttp.responseType = 'blob'
+    xhttp.open('GET', url, true)
+    xhttp.send()
+    
+    xhttp.onload = function() {
+      let fileReader = new FileReader()
+      fileReader.onloadend = function() {
+          newItem = fileReader.result
+          let done = false
+          if (count === 0) done = true
+          callback(newItem, done)         
+      }
+      fileReader.readAsDataURL(xhttp.response)     
+    }
+  }
+
+  convertField (questions, question, date, courseId, count, content) {
+    let newContent = ''
+    content.forEach(item => {
+      let newItem = item
+      if (item.search('src=') !== -1 && item.search('data') === -1) { // convert image to data uri if image source is a url
```

Comment:
Don't split by space when creating content.
Search for < (*s) img  and the closing>
Then search for src(*s) = (*s) and take the content inside of the quotation marks (so, search for the quotation marks as well - single or double quotation marks! ) BUT if the src is already equal to base64, you don't need to do anything!!!